fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatPermissionEvaluator.java

```diff
@@ -83,4 +91,65 @@ private boolean isAuthorized(String username, Resource resource, String resource
     }
     return true;
   }
+
+  @SuppressWarnings("unused")
+  public boolean storeWholePermission() {
+    if (!Boolean.valueOf(fiatEnabled)) {
+      return true;
+    }
+
+    String username = getUsername(SecurityContextHolder.getContext().getAuthentication());
+
+    UserPermission.View view;
+    try {
+      view = fiatService.getUserPermission(username);
+    } catch (RetrofitError re) {
+      String message = String.format("Cannot get whole user permission for user %s", username);
+      log.debug(message);
+      log.trace(message, re);
+      return false;
+    }
+
+    PreAuthenticatedAuthenticationToken auth = new PreAuthenticatedAuthenticationToken(username, null, null);
+    auth.setDetails(view);
+
+    SecurityContext ctx = SecurityContextHolder.createEmptyContext();
+    ctx.setAuthentication(auth);
+    SecurityContextHolder.setContext(ctx);
+
+    return true;
+  }
+
+  private boolean isWholePermissionStored(Authentication authentication) {
+    return authentication.getDetails() != null &&
+        authentication.getDetails() instanceof UserPermission.View;
+  }
+
+  private boolean permissionContains(Authentication authentication,
+                                     String resourceName,
+                                     Resource resource,
+                                     Authorization authorization) {
+    UserPermission.View permission = (UserPermission.View) authentication.getDetails();
+
+    Function<Set<? extends Authorizable>, Boolean> containsAuth = resources ->
+        resources
+            .stream()
+            .anyMatch(view -> view.getName().equalsIgnoreCase(resourceName) &&
+                view.getAuthorizations().contains(authorization));
+
+
+    switch (resource) {
+      case ACCOUNT:
+        return containsAuth.apply(permission.getAccounts());
+      case APPLICATION:
+        return containsAuth.apply(permission.getApplications());
+      default:
+        return false;
+    }
+  }
+
+  @SuppressWarnings("unused")
+  public boolean isAdmin() {
+    return true; // TODO(ttomsu): Chosen by fair dice roll. Guaranteed to be random.
```

Comment:
What will actually be used to determine admin privileges (in future)? Certain group membership?